chunky/src/java/se/llbit/chunky/renderer/RenderWorker.java

```diff
@@ -195,19 +192,17 @@ private void work(RenderTile tile) {
             ray.color.w = 1;
           }
 
-          samples[(y * width + x) * 3 + 0] = ray.color.x;
-          samples[(y * width + x) * 3 + 1] = ray.color.y;
-          samples[(y * width + x) * 3 + 2] = ray.color.z;
+          samples.setPixel(x,y,ray.color.x,ray.color.y,ray.color.z);
 
           scene.finalizePixel(x, y);
 
           if (firstFrame) {
             if (y % 2 == 0 && x < (width - 1)) {
               // Copy the current pixel to the next one.
-              scene.copyPixel(y * width + x, 1);
+              scene.copyPixel(x, y, 1);
```

Comment:
:bulb: In the future we could add a `PreviewSampleBuffer` that supports not having a sample for every pixel and uses the color of the nearest neighbor sample to create an image. That would solve #736